apps/server/src/modules/user-login-migration/service/common-user-login-migration.service.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { roleFactory, schoolDOFactory, setupEntities, userFactory } from '@shared/testing';
+import { Permission, RoleName, SchoolDO, User, UserLoginMigrationDO } from '@shared/domain';
+import { Action, AuthorizationContext, AuthorizationService } from '@src/modules/authorization';
+import { SchoolService } from '@src/modules/school';
+import { CommonUserLoginMigrationService } from './common-user-login-migration.service';
+import { UserLoginMigrationService } from './user-login-migration.service';
+
+describe('CommonUserLoginMigrationService', () => {
+	let module: TestingModule;
+	let service: CommonUserLoginMigrationService;
+
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let userLoginMigration: DeepMocked<UserLoginMigrationService>;
+	let schoolService: DeepMocked<SchoolService>;
+
+	beforeAll(async () => {
+		await setupEntities();
+
+		module = await Test.createTestingModule({
+			providers: [
+				CommonUserLoginMigrationService,
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: UserLoginMigrationService,
+					useValue: createMock<UserLoginMigrationService>(),
+				},
+				{
+					provide: SchoolService,
+					useValue: createMock<SchoolService>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(CommonUserLoginMigrationService);
+		authorizationService = module.get(AuthorizationService);
+		userLoginMigration = module.get(UserLoginMigrationService);
+		schoolService = module.get(SchoolService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('ensurePermission is called', () => {
+		describe('when context, user and school are given ', () => {
+			const setup = () => {
+				const adminRole = roleFactory.buildWithId({
+					name: RoleName.ADMINISTRATOR,
+					permissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+				});
+
+				const school = schoolDOFactory.buildWithId();
+				const user: User = userFactory.buildWithId();
+				const context: AuthorizationContext = {
+					action: Action.write,
+					requiredPermissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+				};
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				schoolService.getSchoolById.mockResolvedValue(school);
+				authorizationService.checkPermissionByReferences.mockResolvedValue(Promise.resolve());
+
+				return {
+					user,
+					schoolId: school.id as string,
```

Comment:
school.id as string looks wrong, can you please remove the "as string" in the whole file?